### ISSUE TITLE
[PROPOSAL] 3DMG Close Ground Animations, Additional Abilities, and Side-Ways Wall-Running

### ISSUE THREAD

**Issue by Prolonpo** [AoTTG-2/AoTTG-2#325]
**How will this feature improve the game?**
Additional features which will permit a greater range of movements to AoTTG2 which will make the game more fun to play as sliding on the ground or hitting obstacles would mean having to restart from 0 speed again.

Ideas in this proposal are from @Pllumped, @antigasp, @desu, @IRI, and @kshz

**Describe the solution you'd like**

(Part 1) - Close Ground Flying Animation

Pllumped:
• When flying close to the ground, the character would adopt a more horizontal position where they will fly as if they are lying on their back or on their sides. The position would be determined by where the hooks are positioned. If above the character, the character would be lying on their back. If to the right of the character, the character would be lying on their right. If to the left of the character, the character would be lying on their left.
• When pressing "dodge" whilst flying in this position, the character would be able to use their arms or their legs to "flip over" obstacles which might have otherwise stopped their flight
• When pressing "dodge" and there are no obstacles nearby, the character would spin horizontally which would inherently give the character an "i-frame" (basically a few frames where the character would be invincible) against AHSS shots and would definitely give the character tons of style
• Ability to hold down the gas burst animation to get a more consistent stronger flow of gas mid-air, this would allow faster acceleration through the air and the ability to quickly dodge incoming attacks at the cost of depleting gas quickly

antigasp:
• From antigasp: during slides, jumping and dodging will preserve whatever momentum is left.

Reference post: https://discord.com/channels/681641241125060652/683107211983323164/777183178263363584

(Part 2) - Ability to jump roof to roof without using gas, revamp on the sliding and collision aspect of the 3DMG, ability to maintain gas burst

desu:
• Ability to jump from roof to roof without using gas
• When pressing the dodge or the jump button during sliding will allow the character to jump and keep whatever speed they had prior to the dodge or jump
• The ability to press "dodge" when close to an obstacles where the characters would be able to "flip over" obstacles which might have otherwise stopped their flight. I'm going to add onto this point your idea that instead of simply bumping into the object, pressing dodge right before the impact would allow the player to use the obstacle as a spring block in order to propel them in the direction where their camera is facing.

IRI:
• Having the ability to have maintained gas burst. @IRI also suggested that this would allow to increase one's flight speed by consuming more gas 

Reference post: https://discord.com/channels/681641241125060652/683107211983323164/777188024541773825

From @KSHZ:
• The sustained gas burst will be able to let players go in the direction that their camera is facing and will also be able to let players gain vertical height or to quickly accelerate towards the ground if needed.

Reference post: https://discord.com/channels/681641241125060652/681647375336341522/777262585870680085

Some additional detail regarding "wall-running" by @desu (link: https://discord.com/channels/681641241125060652/683107211983323164/777300129043578881)
• When running on the wall, the character will continue running until the direction contrary to the wall has been pressed
• If the direction contrary to the wall has not been pressed, the character would only run parallel to the wall for a certain amount of time before automatically jumping of it
• If one has a speed higher than 0.5k or 500u/s, the character upon pressing the dodge button will be able to use the obstacle as a spring board and jump in the direction of the camera whilst also preserving speed
• When pressing the dodge button in mid-air as a character is approaching a wall, they will be able to perform a wall-jump like in Sonic or in Super Mario (see for more detail: https://discord.com/channels/681641241125060652/683107211983323164/777304695944380507)

**Describe alternatives you've considered**
N/A

**Contributor instructions** (Do not modify)
1. Determine if the feature is compatible with existing features and fits within the scope/design of the project
2. If the idea is denied, please provide a reason and close the issue. If the idea is accepted, consult your team to do an estimation of effort. Remove the [PROPOSAL] name and label.
3. Move the issue to the Future Milestone. On rare exceptions an issue may be moved to Release if there's sufficient upvotes.
4. Once the issue matches the current milestone, follow development procedures.

**Comment by Lithrun** [comment]
Seems to be partially related to: #192

**Comment by Prolonpo** [comment]
Indeed! Although in this case, it would be about movement instead of attacks.

**Comment by Prolonpo** [comment]
Precision brought by @Acros5542, the obstacle dodging skill should have a small startup time https://discord.com/channels/681641241125060652/683107211983323164/808177718151479296

But it looks like that overall, it wouldn't make sense as it is debatable if it is even possible to jump off a platform at high speeds...

**Comment by Prolonpo** [comment]
From @Mollymawk, https://discord.com/channels/681641241125060652/681647375336341522/805188789316223057

>Make the player run alongside the wall instead of running up, or if possible give the option for both

@Double Scar zukko aottg2 bot can also be semi-credited with this idea - however the explanation they provided was extremely poor and not explicit at all (link to proposal: https://discord.com/channels/681641241125060652/681647375336341522/698062377282830336 // link to a poorly drawn visual aid: https://discord.com/channels/681641241125060652/683107211983323164/698066698112335902)

**Comment by Alex-Mod-Bot152** [comment]
From @Witched Saea

Better slide mechanic/s 

1) If the angle is shallow enough and you are hooked somewhere you are going to enter similar slide to this one https://cdn.discordapp.com/attachments/683107211983323164/817375724809224272/tumblr_5c83c3a4e4cb6be612a37d4590f5d153_d9a5db5b_640.gif 

2) You are able to gain speed speed until certain point (Around 1k speed, possibly 1,4k) but the acceleration is slower than that of the in the air. If you enter that state beyond that speed you will lose speed slower than with normal slide

3) You can jump from that slide and speed dictates the angle at which you will do that. On lower speed you jump more upwards, on higher speed you jump more forward. If done quickly enough you can exit sliding without losing majority of your speed that you had in the air but 1) has to be met

4) Those things could apply to rooftops to make moving in City less tedious but still requiring skill to pull of successfully. Higher speed is make you slide and bounce of the roof further and higher 

5) If hook/s are not in-front of you will begin to turn slowly in direction of your hooks, like already seen in the Original game 

6) If you reel in during that state you go into normal sliding but if you jump and reel in nothing unusual happens, you simply reel in 

7) That state can be entered from normal sliding if 1) is met


*Late edit. This idea could make horses useless so I think there should be higher gas consumption while going on the ground

**Comment by Prolonpo** [comment]
Child issue #632

**Comment by Prolonpo** [comment]
Additional discussion required was the concluding remark during the 2nd proposal meeting.

**Comment by Prolonpo** [comment]
From **desu** in the staff-ideas-discussion (https://discord.com/channels/681641241125060652/889893004687601724/909902410846191636)
- Vertical gas bursting can be done with a combo of "Dodge whilst mid air + Gas burst"
